quarkchain/cluster/shard_state.py

```diff
@@ -1995,4 +2015,29 @@ def get_total_balance(
         limit: int,
         starter: Optional[bytes] = None,
     ) -> Tuple[int, bytes]:
-        return 42, b""
+        """
+        The starter address should be exclusive in the iteration.
+        """
+        evm_state = self._get_evm_state_from_hash(block_hash)
+        if not evm_state:
+            # TBD, may do hash validation before
+            return 0, b""
+        # the secure trie stored the pair of hash to key
+        # the trie stored the pair of hash to node
+        trie = evm_state.trie.trie
+        if starter:
+            if not evm_state.db.get(utils.sha3_256(starter)):
+                raise Exception(
```

Comment:
same. either return None or raise exception, but the callsite needs to handle it, otherwise the cluster will stop